lib/src/widget.dart

```diff
@@ -138,11 +149,17 @@ class RenderStatefulWidget extends RenderParent<StatefulWidget> {
     if (!identical(configuration, newConfiguration)) {
       // Build the new configuration and decide whether to reuse the child node
       // or replace with a new one.
+      // If there is an existing configuration, call it's #willUnmount lifecycle.
+      _state?.willUnmount();
       _state = newConfiguration.createState();
+      // After creating the new state, call the #willMount lifecycle.
+      _state.willMount();
       _state._config = newConfiguration;
       internalSetStateNode(_state, this);
       Node newChildConfiguration = _state.build();
-      if (_child != null && identical(newChildConfiguration.runtimeType, _child.configuration.runtimeType)) {
+      if (_child != null &&
+          identical(newChildConfiguration.runtimeType,
+              _child.configuration.runtimeType)) {
```

Comment:
This is outside the scope of this PR to fix, but I think this condition is buggy. If two configurations are both an `Element` it will allow a `<div>` to be updated by configuration for a `<span>`. This should be more like `_child.canUpdateFrom(newChildConfiguration)` with `RenderNode` implementations deciding what they can and cannot update from.